engine/include/cubos/engine/render/shadows/directional_caster.hpp

```diff
@@ -19,10 +23,52 @@ namespace cubos::engine
     {
         CUBOS_REFLECT;
 
+        static constexpr int MaxSplits = 10;
+
         cubos::engine::ShadowCaster baseSettings;
 
-        int numSplits;                     ///< Number of splits for PSSM/CSM.
-        std::vector<float> splitDistances; ///< Distance of each split from 0 to 1.
-        float maxDistance = 0;             ///< Max distance (if 0, uses camera zFar).
+        int configNumSplits = 3; ///< Target number of splits for PSSM/CSM.
+        int numSplits = 0;       ///< Current number of splits for PSSM/CSM.
```

Comment:
I would swap these names, and make the 'current number of splits' a private variable.
You could then have a method which compares these two, and returns whether they're different. If they're different, the current is set to the target.